### PR TITLE
Check for instance termination using autoscaling/target-lifecycle-state

### LINKED ISSUE #1: Handle Autoscaling Lifecycle states
New in March '22: https://aws.amazon.com/about-aws/whats-new/2022/03/amazon-ec2-auto-scaling-lifecycle-instance-metadata/

https://docs.aws.amazon.com/autoscaling/ec2/userguide/lifecycle-hooks.html

Would be good to handle these same state changes here - to be notified if a instance is dying not because of a spot being killed, but because the instance is going down. I don't think it makes sense to have a different node module for this?

### PR DESCRIPTION
**Pull request by wbecker**
Fixes #1

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,9 @@ spot.once("termination-cancelled", date => {
 spot.on("termination-not-detected", error => {
     console.log("termination-not-detected", error);
 });
+spot.on("instance-termination", error => {
+    console.log("termination-not-detected", error);
```
Comment (brendtumi) [inline README.md]: can you update the text as "instance-termination" in example.

**Review by brendtumi — CHANGES_REQUESTED** [review]
Thank you very much, if you can update the example we can merge it.

File: README.md (hunk 1 above)
Comment (wbecker) [reply README.md]: Good spot :)

**Comment by wbecker** [comment]
Not 100% sure this is working to be honest - had this running on an instance overnight and it didn't seem to pick it up. It seems like it should be working so it could just be my code that is using it! I'll test it for another day or two then when I'm happy I'll poke you and ask you to merge!

**Comment by brendtumi** [comment]
> Not 100% sure this is working to be honest - had this running on an instance overnight and it didn't seem to pick it up. It seems like it should be working so it could just be my code that is using it! I'll test it for another day or two then when I'm happy I'll poke you and ask you to merge!

Thanks, I will wait until you got a successful result.

**Comment by wbecker** [comment]
It had been failing on my server because the new code was trying to get the results from 2 separate endpoints. Previously, when not in a terminating state the code would get a 404 from termination-time endpoint and error silently. The code in this PR now gets data from two endpoints, so it can't just throw if one of the endpoint 404s. 

I now handle this condition and it seems like it is working on my server now. I'll let it run for 24 hours and make sure there are no further issues, then let's merge!

**Comment by wbecker** [comment]
image

Seems good now - you can see the temporary logging I've added on my staging server.

**Comment by brendtumi** [comment]
version 2.1.0 is published, thank you for your contribution.

**Pull request merged**